tracer-spring-boot-starter/src/main/java/org/zalando/tracer/spring/TracerWebMvcAutoConfiguration.java

```diff
@@ -0,0 +1,51 @@
+package org.zalando.tracer.spring;
+
+import org.apiguardian.api.API;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.Ordered;
+import org.zalando.tracer.Tracer;
+import org.zalando.tracer.servlet.TracerFilter;
+
+import javax.servlet.Filter;
+
+import static javax.servlet.DispatcherType.ASYNC;
+import static javax.servlet.DispatcherType.REQUEST;
+import static org.apiguardian.api.API.Status.INTERNAL;
+import static org.apiguardian.api.API.Status.STABLE;
+
+@API(status = STABLE)
+@Configuration
+@ConditionalOnClass({Tracer.class, Filter.class, FilterRegistrationBean.class})
+@ConditionalOnWebApplication
+@AutoConfigureAfter(name = {
+        "org.springframework.boot.autoconfigure.web.WebMvcAutoConfiguration", // Spring Boot 1.x
+        "org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration" // Spring Boot 2.x
+})
+@Import({TracerAutoConfiguration.class})
+public class TracerWebMvcAutoConfiguration {
```

Comment:
This class needs to be a static inner class of `TracerAutoConfiguration`.

Reply:
I pushed the requested changes. Proper testing is not possible using this setup so I reverted to the initial tests.